### ISSUE TITLE
Micrometer metrics missing

### ISSUE THREAD

**Issue by vauvenal5** [spring-projects/spring-batch#4508]
**Bug description**
According to the documentation the following list of metrics is supported by Spring Batch: https://docs.spring.io/spring-batch/reference/monitoring-and-metrics.html

In the following commit, however, when introducing tracing support the metrics for **job** and **step** were dropped: [e917d75d](https://github.com/spring-projects/spring-batch/commit/e917d75d206816778936c219757c3e80c35af9e9#diff-ce1eecd69e9501d54ab2c7f66ec5e1b66d3b03cd00ee440c762e7fe372509b7d) 

Also I was not able to find the **step.active** metric.

**Expected behavior**
So basically my question, is the documentation wrong or is this a bug? Or am I missing something entirely?

**Comment by vauvenal5** [comment]
Sorry for that it was an issue on our side.

TIL: If auto-config is disabled you have to set the ObservationRegistry at the JobBuilder or StepBuilder to get this metrics generated out of the job and step observations.
